### ISSUE TITLE
Heuristics Segmentation Fault

### ISSUE THREAD

**Issue by robotastic** [szechyjs/dsd#33]
Hi - I am getting a segmentation fault from the new heuristics error correction. I have incorporated DSD into a gnuradio block and I am running several instance in parallel, so it could easily be that. However, I think there might be an error in the DSD code.

In p25p1_tdulc.c, on line 291 read_zeros gets called with new sequence set to zero. 

As a result, sequence_broken for the first element of analog_signal_array is 0 on line 194.

Next contribute_heuristics is called. That brings us to line 186 of p25p1_heuristics.c.

```
prev_dibit = analog_signal_array[i-1].corrected_dibit;
```

This is in a for loop that sets i from 0 to 10.  I am getting the segmentation fault when i=0 because a crazy value is pull from analog_signal_array is pulling a crazy value out and setting prev_dbit to it. This then gets passed to update_p25_heuristics and prev_dbit is used to pull the symbol heuristics:

```
sh = &(heuristics->symbols[previous_dibit][dibit]);
```

In short, it seems like in p25p1_tdulc.c, on line 291 new sequence should be set to 1. I made the switch and it seems to prevent the seg fault.

Also in p25p1_tdu.c on line 21, read_zeros is called with new sequence set to 1.
